src/test/harness.rs

```diff
@@ -52,7 +59,13 @@ impl Environment {
                 .get_old_cwd()
                 .to_path_buf()
         };
-        let project_path = cur_dir.join("test_data").join(project_dir);
+        let project_path = {
```

Comment:
Could we somehow move some a bit of the temp dir logic to `new`? It's not immediately clear to me that if we're passing absolute path then we're using it, otherwise also use the intermediate `test_data` directory, so that may also probably clear that